localization/pose_estimator/ndt_scan_matcher/ndt_omp/CMakeLists.txt

```diff
@@ -1,70 +1,61 @@
-cmake_minimum_required(VERSION 3.0.2)
+cmake_minimum_required(VERSION 3.5)
 project(ndt_omp)
 
-# -mavx causes a lot of errors!!
-add_definitions(-std=c++14 -msse -msse2 -msse3 -msse4 -msse4.1 -msse4.2)
-set(CMAKE_CXX_FLAGS "-std=c++14 -msse -msse2 -msse3 -msse4 -msse4.1 -msse4.2")
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
 
 # warn performance issue of ndt_omp which is NOT built in release mode
-string(TOUPPER ${CMAKE_BUILD_TYPE} uppercase_CMAKE_BUILD_TYPE)
-if (NOT (${uppercase_CMAKE_BUILD_TYPE} STREQUAL "RELEASE"))
-  message(WARNING
-    "CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} may cause a serious performance degradation. Please configure with -DCMAKE_BUILD_TYPE=Release.")
+if (NOT ("${CMAKE_BUILD_TYPE}" STREQUAL "Release"))
+  message(WARNING "CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} may cause a serious performance degradation. Please configure with -DCMAKE_BUILD_TYPE=Release.")
 endif()
 
-find_package(catkin REQUIRED COMPONENTS
-  roscpp
-  pcl_ros
-)
-
+find_package(ament_cmake REQUIRED)
+find_package(ament_cmake_auto REQUIRED)
+find_package(PCL REQUIRED)
 find_package(OpenMP)
+
 if (OPENMP_FOUND)
   set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
   set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
 endif()
 
-catkin_package(
-  INCLUDE_DIRS include
-  LIBRARIES ndt_omp
-  CATKIN_DEPENDS
-)
-
 ###########
 ## Build ##
 ###########
 
-include_directories(
-  include
-  ${catkin_INCLUDE_DIRS}
-)
-
 add_library(ndt_omp
   src/ndt_omp/voxel_grid_covariance_omp.cpp
   src/ndt_omp/ndt_omp.cpp
 )
 
-add_executable(align
-  apps/align.cpp
-)
-add_dependencies(align
-  ndt_omp
+target_include_directories(ndt_omp
+  PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include> 
 )
+
 # pcl 1.7 causes a segfault without -O2 (or -O3) flag
-target_compile_options(ndt_omp PUBLIC
-  $<$<CONFIG:Debug>:-O2 -g>
-)
-target_link_libraries(align
-  ${catkin_LIBRARIES}
-  ndt_omp
+target_compile_options(ndt_omp
+  PUBLIC
+    $<$<CONFIG:Debug>:-O2 -g>
+  PRIVATE
+    # -mavx causes a lot of errors!!
+    -msse -msse2 -msse3 -msse4 -msse4.1 -msse4.2
 )
 
-install(TARGETS ndt_omp
-  ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-  RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
+set(NDT_OMP_DEPENDENCIES
+  PCL
+  OpenMP
 )
 
-## Install project namespaced headers
-install(DIRECTORY include/${PROJECT_NAME}/
-  DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
+ament_target_dependencies(ndt_omp ${NDT_OMP_DEPENDENCIES}) 
+
+
+add_executable(align
+  apps/align.cpp
 )
+
+target_link_libraries(align ndt_omp ${PCL_LIBRARIES})
```

Comment:
I have to write `${PCL_LIBRARIES}` here, `ament_target_dependencies(align ${NDT_OMP_DEPENDENCIES})` doesn't work even though it appears to work in https://github.com/tier4/Pilot.Auto/pull/21/files. I would prefer to use `ament_target_dependencies` since it also takes `PCL_LIBRARY_DIRS`, `PCL_INCLUDE_DIRS` etc. into account and is shorter.

@fred-apex-ai Any idea why?

Reply:
@nnmm did our discussion solve this issue?